rock/src/io_stream.mli

```diff
@@ -0,0 +1,17 @@
+module Input : sig
+  type 'a t
+
+  val create : (unit -> 'a option Lwt.t) -> 'a t
+  val singleton : 'a -> 'a t
+  val read : 'a t -> 'a option Lwt.t
+  val iter : ('a -> unit Lwt.t) -> 'a t -> unit Lwt.t
+end
+
+module Output : sig
+  type 'a t
+
+  val create : ('a option -> unit Lwt.t) -> 'a t
+  val write : 'a option -> 'a t -> unit Lwt.t
+end
+
+val transfer : 'a Input.t -> 'a Output.t -> unit Lwt.t
```

Comment:
I'd try to match the API here: https://github.com/ocaml/ocaml-lsp/blob/master/fiber-unix/src/fiber_stream.mli#L33

It's already the one I'm familiar with.

Reply:
Good idea. I'll add in the Lwt version of fiber_stream module.